core/module.h

```diff
@@ -53,6 +52,24 @@ using bess::gate_idx_t;
 #define MAX_TASKS_PER_MODULE 32
 #define UNCONSTRAINED_SOCKET ((0x1ull << MAX_NUMA_NODE) - 1)
 
+struct Context {
```

Comment:
I wonder if we should make this more strict about access to internal fields (using class instead of struct, and private or protected members with acccessors).  I'm not exactly militant about it though, perhaps from too much old school C programming 😈

Reply:
Definitely, I can change it to class.